src/main/scala/mesosphere/marathon/upgrade/DeploymentActor.scala

```diff
@@ -145,40 +143,41 @@ private class DeploymentActor(
     status: DeploymentStatus): Future[Unit] = {
     logger.debug("Scale runnable {}", runnableSpec)
 
-    val runningInstances = instanceTracker.specInstancesSync(runnableSpec.id).filter(_.state.condition.isActive)
-
     def killToMeetConstraints(notSentencedAndRunning: Seq[Instance], toKillCount: Int) = {
       Constraints.selectInstancesToKill(runnableSpec, notSentencedAndRunning, toKillCount)
     }
 
-    val ScalingProposition(tasksToKill, tasksToStart) = ScalingProposition.propose(
-      runningInstances, toKill, killToMeetConstraints, scaleTo, runnableSpec.killSelection)
-
-    def killTasksIfNeeded: Future[Unit] = {
+    def killTasksIfNeeded(tasksToKill: Option[Seq[Instance]]): Future[Unit] = {
       logger.debug("Kill tasks if needed")
       tasksToKill.fold(Future.successful(())) { tasks =>
         logger.debug("Kill tasks {}", tasks)
         killService.killInstances(tasks, KillReason.DeploymentScaling).map(_ => ())
       }
     }
 
-    def startTasksIfNeeded: Future[Unit] = {
-      tasksToStart.fold(Future.successful(())) { tasksToStart =>
-        logger.debug(s"Start next $tasksToStart tasks")
+    def startTasksIfNeeded(tasksToStart: Option[Int]): Future[Unit] = {
+      tasksToStart.fold(Future.successful(())) { tasks =>
+        logger.debug(s"Start next $tasks tasks")
         val promise = Promise[Unit]()
         context.actorOf(TaskStartActor.props(deploymentManager, status, driver, scheduler, launchQueue, instanceTracker, eventBus,
           readinessCheckExecutor, runnableSpec, scaleTo, promise))
         promise.future
       }
     }
 
-    killTasksIfNeeded.flatMap(_ => startTasksIfNeeded)
+    val futureRunningInstances = instanceTracker.specInstances(runnableSpec.id).map(_.filter(_.state.condition.isActive))
+    futureRunningInstances.flatMap({ runningInstances =>
```

Comment:
"({" is redundant, please just use "{".

Also, like the other PRs outstanding, please prefer async/await over monadic methods on Futures.